rust_src/src/windows.rs

```diff
@@ -510,6 +518,22 @@ impl LispWindowRef {
             self.get_frame().config_scroll_bar_height
         }
     }
+
+    /// Return the bottom boundary y-position for text lines in window W.
+    /// This is the first y position at which a line cannot start.
+    /// It is relative to the top of the window.
+    ///
+    /// This is the height of W minus the height of a mode line, if any.
+    pub fn text_bottom_y(mut self) -> i32 {
+        self.pixel_height
+            - self.bottom_divider_width()
+            - self.scroll_bar_area_height()
+            - if self.wants_mode_line() {
```

Comment:
I'm pretty sure there isn't a way of doing it in the standard library, but I suppose it would be possible to implement a method on bool that returns one of two values depending on the bools value. Something like `predicate.true_or(value, false_value)`

Reply:
This is a case where I miss C's ternary operator. It's perfectly suited for simple checks like this